### ISSUE TITLE
Style/Attr is broken when an "attr" method is defined

### ISSUE THREAD

**Issue by localhostdotdev** [rubocop-hq/rubocop#7029]
if I have a method called `attr`, e.g. `attr(*args, &block)` or something like that then it breaks this cop.

--------

## Expected behavior

maybe it could only inspect classes/modules body?

## Actual behavior

inspects methods body

## Steps to reproduce the problem

source:

```ruby
class A
  def attr(*args)
    p args
  end

  def a
    attr(1)
    attr(:a, 1)
    attr(:b) { 2 }
    attr(:c) do |a|
      2
    end
  end
end
```

auto-corrected:

```ruby
class A
  def attr(*args)
    p args
  end

  def a
    attr_reader(1)
    attr_reader(:a, 1)
    attr_reader(:b) { 2 }
    attr_reader(:c) do |_a|
      2
    end
  end
end
```

(not sure how but I also had some `attr_accessor` and `attr_writer` inserted in my code, it somehow generated invalid ruby even like `attr_writer(:something`)

## RuboCop version

```
$ rubocop -V
0.68.1 (using Parser 2.6.3.0, running on ruby 2.6.3 x86_64-darwin18)
```

**Comment by localhostdotdev** [comment]
also messes up with `Lint/DuplicateMethods`
